src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/Writer.java

```diff
@@ -176,7 +244,7 @@ private void updateResponseHeaders(
                     : "stale-while-revalidate=" + staleWhileRevalidate;
             builder.item(header -> header.name("cache-control").value(value));
         }
-        if (!responseHeadersRO.anyMatch(h -> ETAG.equals(h.name().asString())))
+        if (!responseHeadersRO.anyMatch(h -> ETAG.equals(h.name().asString())) && etag != null)
```

Comment:
Avoid using `responseHeadersRO` in local variable names, the `RO` and `RW` suffixes are for the member fields only.